using_clang_format.md

```diff
@@ -0,0 +1,69 @@
+# Using clang-format
```

Comment:
Don't introduce a new document that isn't linked from anywhere. Place this in "COMPILING" or at least add some links from that file to this one. In short: don't make it harder for people to contribute.

Reply:
Moved this into COMPILING.md